src/Components/Blazor/Server/src/MonoDebugProxy/BlazorMonoDebugProxyAppBuilderExtensions.cs

```diff
@@ -52,6 +54,66 @@ public static void UseBlazorDebugging(this IApplicationBuilder app)
             });
         }
 
+        private static void UseVisualStudioDebuggerConnectionRequestHandlers(this IApplicationBuilder app)
+        {
+            // Unfortunately VS doesn't send any deliberately distinguishing information so we know it's
+            // not a regular browser or API client. The closest we can do is look for the *absence* of a
+            // User-Agent header. In the future, we should try to get VS to send a special header to indicate
+            // this is a debugger metadata request.
+            app.Use(async (context, next) =>
+            {
+                var request = context.Request;
+                var requestPath = request.Path;
+                if (requestPath.StartsWithSegments("/json")
+                    && !request.Headers.ContainsKey("User-Agent"))
+                {
+                    var debuggerHost = "http://localhost:9222";
```

Comment:
Eventually we might need to make this port number (and possibly hostname) configurable, but hopefully we will only do that if we find out it's necessary for tooling (e.g., if VS really wants to launch the browser with a nonstandard debugging port). So I'm OK with this as-is for now.

Reply:
> Eventually we might need to make this port number (and possibly hostname) configurable

Wouldn't this fit in `launchSettings`? I don't know if VS actually picks an _unused_ port to launch IIS Express or simply picks any at random (and might randomly fail), but that seems semantically related.

Reply:
@chucker This is the port that the browser listens on for debugging, not the port your app runs on. So the correct value depends on how you (or the IDE) started up your browser.

Reply:
@pranavkm: OK, we've learned now thanks to a meeting with VS and web tools people that we *do* need to make this into a parameter. We're thinking it should be an environment variable. So:

 * If the environment variable ASPNETCORE_WEBASSEMBLYDEBUGHOST is set, we should use that value for `debuggerHost`
 * Otherwise we keep using `http://localhost:9222`

Do you think that's a change we could include in this PR?